src/unistd.rs

```diff
@@ -539,6 +539,23 @@ pub fn symlinkat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+// Double the buffer capacity up to limit. In case it already has
+// reached the limit, return Errno::ERANGE.
+fn reserve_buffer_size<T>(buf: &mut Vec<T>, limit: usize) -> Result<()> {
+    use std::cmp::min;
+
+    if buf.len() == limit {
+        return Err(Error::Sys(Errno::ERANGE))
+    }
+
+    unsafe { buf.set_len(buf.capacity()) };
+
+    let capacity = min(buf.capacity() * 2, limit);
+    buf.reserve_exact(capacity);
```

Comment:
No need to override the standard library.  You can use reserve instead of reserve_exact.  I just don't want to rely on `reserve(1)` actually reserving more space than 1.